packages/number-field/src/NumberField.ts

```diff
@@ -759,5 +770,15 @@ export class NumberField extends TextfieldBase {
             }
             this.inputElement.inputMode = inputMode;
         }
+        if (
+            changes.has('focused') &&
+            this.focused &&
+            !this.hasRecentlyReceivedPointerDown &&
+            !!this.formatOptions.unit
+        ) {
+            // Normalize keyboard focus entry between unit and non-unit bearing Number Fields
+            this.setSelectionRange(0, this.displayValue.length);
```

Comment:
Do we need to check if displayValue has a length or not right?

Reply:
`displayValue` is always a string, and even if it's length is `0` the outcome here will still be correct.